config/file_config.go

```diff
@@ -92,20 +92,21 @@ type AccessKeyConfig struct {
 }
 
 // truncate the key to 8 characters for logging
-func (a *AccessKeyConfig) sanitize(key string) string {
+func (a *AccessKeyConfig) Sanitize(key string) string {
 	return fmt.Sprintf("%.8s...", key)
 }
 
-// CheckAndMaybeReplaceKey checks the given API key against the configuration
-// and possibly replaces it with the configured SendKey, if the settings so indicate.
-// It returns the key to use, or an error if the key is invalid given the settings.
-func (a *AccessKeyConfig) CheckAndMaybeReplaceKey(apiKey string) (string, error) {
-	// Apply AcceptOnlyListedKeys logic BEFORE we consider replacement
-	if a.AcceptOnlyListedKeys && !slices.Contains(a.ReceiveKeys, apiKey) {
-		err := fmt.Errorf("api key %s not found in list of authorized keys", a.sanitize(apiKey))
-		return "", err
+func (a *AccessKeyConfig) IsAccepted(key string) bool {
```

Comment:
I think this could be a little clearer, maybe something like `IsValid` instead? We should add a function description too as it's public function.

Reply:
Also, we create similar errors when this returns false so we could update this func to return the error and check if it's nil when calling it.

Reply:
I changed the function signature to return an error which eliminated a lot of duplicated code. Hooray! 

I'm a bit hesitant to name this function `IsValid` since the key could be a very valid api key but it's just not in the accepted key list. I would like to keep the name matching with what the config is doing